packages/protocol/scripts/truffle/make-release.ts

```diff
@@ -0,0 +1,244 @@
+      // before or after other contracts in this script?), and that ordering is not being
+      // checked, fail if there are storage incompatible changes to Governance.
+      if (contractName === 'Governance') {
+        throw new Error(`Storage incompatible changes to Governance are not yet supported`)
+      }
+      console.log(`Deploying ${contractName}Proxy`)
+      const Proxy = await artifacts.require(`${contractName}Proxy`)
+      // Hack to trick truffle, which checks that the provided address has code
+      const proxy = await (argv.dry_run ? Proxy.at(REGISTRY_ADDRESS) : Proxy.new())
+      const initializeAbi = (contract as any).abi.find(
+        (abi: any) => abi.type === 'function' && abi.name === 'initialize'
+      )
+      console.log(`Setting ${contractName}Proxy implementation to ${contract.address}`)
+      if (initializeAbi) {
+        const args = initializationData[contractName]
+        console.log(`Initializing ${contractName} with: ${args}`)
+        if (!argv.dry_run) {
+          await setAndInitializeImplementation(
+            web3,
+            proxy,
+            contract.address,
+            initializeAbi,
+            {},
+            ...args
+          )
+        }
+      } else {
+        if (!argv.dry_run) {
+          await proxy._setImplementation(contract.address)
+        }
+      }
+      // This makes essentially every contract dependent on Governance.
+      console.log(`Transferring ownership of ${contractName}Proxy to Governance`)
+      if (!argv.dry_run) {
+        await proxy._transferOwnership(addresses.get('Governance'))
+      }
+      const proxiedContract = await artifacts.require(contractName).at(proxy.address)
+      const transferOwnershipAbi = (contract as any).abi.find(
+        (abi: any) => abi.type === 'function' && abi.name === 'transferOwnership'
+      )
+      if (transferOwnershipAbi) {
+        console.log(`Transferring ownership of ${contractName} to Governance`)
+        if (!argv.dry_run) {
+          await proxiedContract.transferOwnership(addresses.get('Governance'))
+        }
+      }
+      return proxy
+    }
+
+    const release = async (contractName: string) => {
+      if (released.has(contractName)) {
+        return
+      } else {
+        // 1. Release all dependencies.
+        const contractDependencies = dependencies.get(contractName)
+        for (const dependency of contractDependencies) {
+          await release(dependency)
+        }
+        // 2. Link dependencies.
+        const Contract = await artifacts.require(contractName)
+        await Promise.all(contractDependencies.map((d) => Contract.link(d, addresses.get(d))))
+
+        // This is a hack that will re-deploy all libraries with proxies, whether or not they have
+        // changes to them.
+        // TODO(asa): Remove `isLibrary` for future releases.
+        const isLibrary = Object.keys(linkedLibraries).includes(contractName)
```

Comment:
Can you not just directly check against the key here with `linkedLibraries[contractName]`?

Reply:
You tell me!